LayoutTests/accessibility/text-marker/text-marker-bidi-element-arabic.html

```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
+<html>
+<head>
+    <meta charset="utf-8">
+    <script src="../../resources/js-test.js"></script>
+    <script src="../../resources/accessibility-helper.js"></script>
+    <style>
+        .unicode-bidi-normal {
+            unicode-bidi: normal;
+        }
+        .unicode-bidi-embed {
+            unicode-bidi: embed;
+        }
+        .unicode-bidi-bidi-override {
+            unicode-bidi: bidi-override;
+        }
+        .unicode-bidi-plaintext {
+            unicode-bidi: plaintext;
+        }
+        .unicode-bidi-isolate {
+            unicode-bidi: isolate;
+        }
+        .unicode-bidi-isolate-override {
+            unicode-bidi: isolate-override;
+        }
+    </style>
+</head>
+
+<body>
+<div id="content-normal" class="unicode-bidi-normal">Anyone can try لغة برمجة C++</div>
+<div id="content-embed" class="unicode-bidi-embed">Anyone can try لغة برمجة C++</div>
+<div id="content-bidi-override" class="unicode-bidi-bidi-override">Anyone can try لغة برمجة C++</div>
+<div id="content-plaintext" class="unicode-bidi-plaintext">Anyone can try لغة برمجة C++</div>
+<div id="content-isolate" class="unicode-bidi-isolate">Anyone can try لغة برمجة C++</div>
+<div id="content-isolate-override" class="unicode-bidi-isolate-override">Anyone can try لغة برمجة C++</div>
+<script>
+var output = "This test verifies that a combination of Arabic and English is read correctly by Accessibility, when using CSS unicode-bidi property.\n\n";
+
+if (window.accessibilityController) {
+    var testIDs = ["content-normal", "content-embed", "content-plaintext", "content-isolate"] 
+    var testOverrideIDs = ["content-bidi-override", "content-isolate-override"]
+    var staticText, staticTextTextMarkerRange, startMarker, endMarker;
+    testIDs.forEach(testID => {
+        staticText = accessibilityController.accessibleElementById(testID).childAtIndex(0);
+        staticTextTextMarkerRange = staticText.textMarkerRangeForElement(staticText);
+        output += expect("staticText.textMarkerRangeLength(staticTextTextMarkerRange)", "28");
+
+        output += expectRectWithVariance("staticText.boundsForRange(0, 28)", -1, -1, 188, 18, 1);
```

Comment:
expectRectWithVariance is good, but it still does feel like this test might tend to break if things shift by a couple of pixels.

Would it be possible to instead iterate over the characters and, using the bounds, just assert whether or not each index moved to the left or to the right?